pybatchai.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from azure.common.credentials import ServicePrincipalCredentials
+import click
+import coloredlogs
+
+import cli.resource_group
+import cli.validation
+
+@click.group()
+@click.option('--subscription-id', required=True, 
+              callback=cli.validation.validate_uuid)
+@click.option('--resource-group', required=True, help='resource group name', 
+              callback=cli.validation.validate_rg_name)
+@click.option('--location', required=True, default='eastus',
+              callback=cli.validation.validate_location)
+@click.option('--aad-app-id', required=True, 
+              callback=cli.validation.validate_uuid)
+@click.option('--aad-key', required=True)
+@click.option('--aad-directory-id', required=True,
+              callback=cli.validation.validate_uuid)
+@click.pass_context
+def main(context: object,
+         subscription_id: str,
+         resource_group: str,
+         location: str,
+         aad_app_id: str,
+         aad_key: str,
+         aad_directory_id: str) -> None:
```

Comment:
https://github.com/python/typing/issues/433
this thread has some interesting ideas on how to style cases like this

Reply:
I like this case:
```
def some_example_func(
        app_cfg_file: str,
        tmpl_list_file: str = 'default_list',
        style: str = '',
        input_env: Union[str, None] = None,
        tmpl_name: Union[str, None] = None
    ) -> list:
    """This is some example function."""
```

If we need to add parameters, it should be easy as they'll be on a newline (no merge conflicts). I also like how they're more highlighted because they're on a new line.